src/components/ars.js

```diff
@@ -1,105 +1,132 @@
 /**
  * Ars
  * The main element for Ars Arsenal
+ * @flow
  */
 
-import Picker from './picker'
 import React from 'react'
+import cx from 'classnames'
+import Picker from './picker'
 import Selection from './selection'
 import MultiSelection from './multiselection'
-import Sync from '../mixins/sync'
-import createClass from 'create-react-class'
-import cx from 'classnames'
-import { func } from 'prop-types'
-
-let Ars = createClass({
-  mixins: [Sync],
-
-  propTypes: {
-    onChange: func
-  },
-
-  getDefaultProps() {
-    return {
-      onChange: () => {},
-      rootAttributes: {},
-      multiselect: false,
-      resource: 'Photo',
-      mode: 'gallery'
-    }
-  },
+import { type Record } from '../record'
+
+type Props = {
+  makeQuery: string => string,
+  makeURL: (string, any) => string,
+  rootAttributes: { [string]: * },
+  multiselect: boolean,
+  resource: string,
+  mode: 'gallery' | 'table',
+  picked: string | number | Array<string | number>,
+  columns?: string[],
+  url: string,
+  onFetch: (*) => Record,
+  onError: (*) => *,
+  onChange: (*) => any
+}
```

Comment:
I don't think all of these attributes are necessary, but I need them temporarily until I completely get rid of the mixins for fetching data.